apps/televisit-demo/backend/signinhook/index.js

```diff
@@ -0,0 +1,38 @@
+const AWS = require('aws-sdk');
+
+AWS.config.update({ region: process.env.AWS_REGION });
+const chime = new AWS.Chime({ region: process.env.AWS_REGION });
+
+const { CHIME_APP_INSTANCE_ARN } = process.env;
+
+exports.handler = async (event, context, callback) => {
+  const username = event.userName;
+  const userId = event.request.userAttributes.profile;
+
+  // 'none' is default user profile attribute in Cognito upon registration which
+  if (userId === 'none') {
+    console.log(`User hasn't logged in yet and hasn't been setup with profile`);
+    callback(null, event);
+  }
+  // Create a Chime App Instance User for the user
+  const chimeCreateAppInstanceUserParams = {
+    AppInstanceArn: CHIME_APP_INSTANCE_ARN,
+    AppInstanceUserId: userId,
+    Name: username
+  };
+
+  try {
+    console.log(`Creating app instance user for ${userId}`);
+    await chime
+      .createAppInstanceUser(chimeCreateAppInstanceUserParams)
+      .promise();
+  } catch (e) {
+    console.log(JSON.stringify(e));
+    return {
+      statusCode: 500,
+      body: e.stack
```

Comment:
What if e.stack is null?

You can either add an arbitrary string like `Server error while creating app instance` or add a null check if you want to display the error from the service. Usually, we try to set the error message ourselves to control messaging.